pkg/nodejs/pnpm/parse.go

```diff
@@ -95,15 +116,61 @@ func (p *Parser) parse(lockFile *LockFile) ([]types.Library, []types.Dependency)
 	return libs, deps
 }
 
-func isIndirectLib(name string, directDeps map[string]string) bool {
+func isIndirectLib(name string, directDeps map[string]interface{}) bool {
 	_, ok := directDeps[name]
 	return !ok
 }
 
-func getPackageNameAndVersion(pkg string) (string, string) {
-	idx := strings.LastIndex(pkg, "/")
-	name := pkg[1:idx]
-	version := pkg[idx+1:]
-
+func getPackageNameAndVersion(pkg, name, version string, lockFileVersion float64) (string, string) {
+	// local archives have `name` and `version` fields
+	// https://github.com/pnpm/spec/blob/ad27a225f81d9215becadfa540ef05fa4ad6dd60/lockfile/5.2.md#packagesdependencypathname
+	if strings.HasPrefix(pkg, "file:") {
+		if _, err := semver.Parse(version); err != nil {
+			log.Logger.Debugf("Skip %q package. %q doesn't match semver: %s", pkg, version, err)
+			return "", ""
+		}
+		return name, version
+	}
+	versionSep := "@"
+	if lockFileVersion < 6 {
+		versionSep = "/"
+	}
+	return parsePackage(pkg, versionSep)
+}
+func parsePackage(pkg, versionSep string) (string, string) {
```

Comment:
```suggestion
// cf. https://github.com/pnpm/pnpm/blob/ce61f8d3c29eee46cee38d56ced45aea8a439a53/packages/dependency-path/src/index.ts#L112-L163
func parseDependencyPath(pkg, versionSep string) (string, string) {
```

Reply:
Made in https://github.com/aquasecurity/go-dep-parser/pull/220/commits/2babc94911560f26355064c29a8340c4765ad992